updated with minor adjustments to preprocessing and model params

zbuckley opened the pull request.

Commit:
updated with minor adjustments to preprocessing and model params

zbuckley commented on Code/CheXpertDataset.py:
```diff
@@ -66,14 +66,12 @@ def __call__(self, sample):
     def __call__(self, sample):
         sample = sample[label_columns].replace(-1, 0)
-        sample = sample[label_columns].fillna(0)
         return sample
```
realized these lines aren't needed. see label_processing_uzeros and uones in the common_training file

josh-bernstein618 replied:
Lol I meant to delete them from the other file when I changed this.  But since we're always changing blanks to 0s, this works too!

josh-bernstein618 replied:
Oh wait, you changed it in the other file.  Glad it wasn't an oversight on my part

zbuckley replied:
I only changed it because for some reason it wasn't working before. don't remember why. but the NaN remover worked... so i simply composed them together... it also seems a bit cleaner.

zbuckley commented on Code/training_team8.py:
```diff
@@ -39,9 +40,10 @@
     'batch_size': BATCH_SIZE,
     'shuffle': True,  # why not.. *shrug*?
-    'num_workers': 12}
+    'num_workers': 16}
```
bumped this up for my 16 core machine.

zbuckley commented on Code/training_team8.py:
```diff
@@ -39,9 +40,10 @@
+    'num_workers': 16}
 
 training_loader, validation_loader = build_data_loaders(
+    base_label_transform,
```
added an ability to pass in the label transformer so that the training file controls it. 
Should better facilitate copying training file for each model we're training.

zbuckley commented on Code/training_common_utils.py:
```diff
@@ -141,10 +147,11 @@ def training_loop(
         for i, (images, labels) in enumerate(training_loader):
             print("\tMiniBatch {}:".format(i))
             minibatch_start = time.time()
+            #print("labels:", labels)
```
added this for debugging (was getting nan loss values again. 
then commented it out for more reasonable logging.

zbuckley commented on Code/training_common_utils.py:
```diff
@@ -187,11 +194,11 @@ def training_loop(
         print("\tEpoch Compute Time: {}".format(
             time.time() - epoch_start))
-        if((loss_training_calc < min_train_loss) | (loss_validation_calc < min_val_loss):
+        if loss_training_calc < min_train_loss or loss_validation_calc < min_val_loss:
```
had to adjust the syntax. my python interpreter didn't like it as is.

zbuckley replied:
as a thought... should it really be this instead? I think we may not be getting the desired behavior this way.
```suggestion
        if loss_validation_calc < min_val_loss:
```

kshitijbhat330 replied:
Which version are you using? it works fine with my Python 3. Anyway, this is fine too

zbuckley replied:
python 3.6.. 
i also realized later that we really shouldn't be basing it on loss_training_calc, since that is always  going to decrease (the whole backpropagation algorithm's convergence thing...) as the network "memorizes" the training data.

zbuckley commented on Code/training_team8.py:
```diff
@@ -24,8 +25,8 @@
-BATCH_SIZE = 8
-N_EPOCHS = 50
+BATCH_SIZE = 16
+N_EPOCHS = 15
```
so far. on the full training set. I'm getting ~36 minutes per epoch. 
so 15 * 30 ~= 7.5 hours. If i'm right, I'll be done around 2/3am

zbuckley commented on Code/training_team8.py:
```diff
@@ -24,8 +25,8 @@
 MOMENT = .9
-BATCH_SIZE = 8
-N_EPOCHS = 50
+BATCH_SIZE = 16
```
increased this because my hardware supports it, and it'll speed things up. can be lowered if we run into CUDA ram errors.

Commit:
various changes.
renamed training_team8 to training uzeros
created training_uones
removed bash script content that gives syntax errors from download
scripts
added trained models download script

josh-bernstein618 commented on Code/training_common_utils.py:
```diff
@@ -187,11 +195,11 @@ def training_loop(
             time.time() - epoch_start))
-        if((loss_training_calc < min_train_loss) | (loss_validation_calc < min_val_loss):
+        if loss_validation_calc < min_val_loss:
             print("*******Loss imrpoved*******")
```
```suggestion
            print("*******Loss improved*******")
```

josh-bernstein618 commented on Code/training_team8_uzeros.py:
```diff
@@ -0,0 +1,88 @@
+# Starting from sample code:
```
I am not crazy about having 2 files that do almost the same thing.  The only difference is whether it imports `label_preprocessing_uzeros` or `label_preprocessing_uones` as base_label_transform.    I don't think this really justifies having 2 separate long files.  I am willing to accept this for our efficiency, but I wouldn't be surprised if Amir thinks it's ugly.

zbuckley replied:
Amir will not care. 
That being said my attempt to reduce copy-pasting was the common_training file. I'd intended to start copying the training files per model we need to train. If you want to go with another approach we'll need to set it up, but this should allow us to pursue separate models over the next few days without running into crazy merge conflicts.

josh-bernstein618 reviewed (approved):
LGTM, did not test
